### PR TITLE
Add stacklevel=2 to some more warnings.warn() calls

### PR DESCRIPTION
**Pull request by cldssty**
## PR Summary
Have found some more warnings.warn() calls that have not set stacklevels and have set them all to 2. 

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by cldssty** [comment]
#10643

**Comment by jklymak** [comment]
Great!  Can you squash all these commits to keep the commit history cleaner?

**Comment by jklymak** [comment]
@cldssty a couple of your lines went over 79 characters, which our tests get mad about.  See the output of the travis-ci job above.   Thanks!

**Comment by cldssty** [comment]
ok will fix these in a bit!

**Comment by cldssty** [comment]
going to squash them now

**Comment by ImportanceOfBeingErnest** [comment]
There are still 4 trailing whitespaces somewhere. (You may look at the test results by clicking the "Details" button next to the travis-ci test to find out where exactly.)

**Comment by cldssty** [comment]
Thanks for letting me know about looking at the travis build for what to fix. Currently fixing further problems with this set of commits in my own fork - will squash and make new pull request soon. Thanks!

**Comment by cldssty** [comment]
Will squash commits first thing tmr!

**Comment by cldssty** [comment]
does this mean a test is needed?

**Comment by ImportanceOfBeingErnest** [comment]
No, I don't think you need to test those warnings. But if there is a warning which you think is crucial and where you think this is necessary, you may of course add a test.

**Comment by cldssty** [comment]
I see. I was just wondering whether there is something I could do to fix the code coverage error

**Comment by cldssty** [comment]
omg I think I might have squashed more than I needed!!

**Comment by cldssty** [comment]
have created new PR where I've squashed only the commits I've made!

File: lib/matplotlib/style/core.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def _remove_blacklisted_style_params(d, warn=True):
             if warn:
                 warnings.warn(
                     "Style includes a parameter, '{0}', that is not related "
-                    "to style.  Ignoring".format(key))
+                    "to style.  Ignoring".format(key), stacklevel=2)
```
Comment (timhoffm) [inline lib/matplotlib/style/core.py]: Could be `stacklevel=3` because it's inside a private wrapper.

Comment (jklymak) [reply lib/matplotlib/style/core.py]: @cldssty did you want to make this change?  Prob fine as a separate commit if you don't feel like rebasing...

**Comment by jklymak** [comment]
Thanks a lot @cldssty

**Pull request merged**